include/deal.II/meshworker/copy_data.h

```diff
@@ -0,0 +1,119 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_numerics_copy_data_h
+#define dealii_numerics_copy_data_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/types.h>
+
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace MeshWorker
+{
+  /**
+   * Helper copy data struct.
+   *
+   * This class is a good default drop in CopyData object for the
+   * WorkStream::run() and MeshWorker::mesh_loop() functions.
+   *
+   * It allows you to store arbitrary number of (local) FullMatrices, vectors,
+   * and local_dof_indices vectors.
+   *
+   * @author Luca Heltai, 2019.
+   */
+  template <int n_matrices    = 1,
+            int n_vectors     = n_matrices,
+            int n_dof_indices = n_matrices>
+  struct CopyData
+  {
+    /**
+     * Initialize everything with the same @p size.
+     */
+    CopyData(const unsigned int size)
+    {
```

Comment:
Our usual style is to have the implementation of functions separate from the declaration. Would you be willing to externalize the implementations? I think eclipse even has a function for this.

Reply:
In all fairness, @luca-heltai did say in his initial comment

> Before moving the implemenation to .cc and templates.h...

Reply:
Yes, I suspected something like this. There is also a similar comment at the bottom of one of the files.

Reply:
Yes, I was planning to do that in the last commit, before squashing everyhing.

I don't want to spend hours in linking the library for every little change to this file... :)